src/cev/_embedding.py

```diff
@@ -54,19 +54,18 @@ def _prepare_ozette(df: pd.DataFrame, robust_only: bool = True):
         robust = (df["faustLabels"] != NON_ROBUST_LABEL).to_numpy()
         representative_label = df["faustLabels"][robust].iloc[0]
 
+        coords = df[["umapX", "umapY"]].to_numpy()
+
         if robust_only:
             df = df[robust].reset_index(drop=True)
-            coords = df[["umapX", "umapY"]].to_numpy()
             labels = df["faustLabels"].to_numpy()
             robust = None
         else:
-            coords = df[["umapX", "umapY"]].to_numpy()
-            df = df[["faustLabels"]]
-            df["label"] = ""
+            df["labels"] = ""
             for marker in parse_label(representative_label):
                 marker_annoation = marker.name + df[f"{marker.name}_faust_annotation"]
-                df["label"] += marker_annoation
-            labels = df["label"].to_numpy()
```

Comment:
Thanks for fixing this. I wonder if we would run into a similar issue if the df already had a `labels` column. Should we give `labels` a more unique name to avoid collisions?

Reply:
Do we even need to assign anything to `df` or could we just create a new `pd.Series`? Since we only return `coords, labels, robust` I was wondering why adjusted `df` in the first place or is this a side effect?

Reply:
I agree. We should avoid mutating the input df. The code there currently is just a hack to create a series with a shared index into the data frame that we can mutate in the for loop. I think we could do something like.

```python
label_series = pd.Series("", index=df.index)
```